client/packages/programs/src/JsonForms/components/EncounterProgramEvent.tsx

```diff
@@ -100,25 +101,27 @@ const UIComponent = (props: ControlProps) => {
 
   const encounterId = useEncounter.utils.idFromUrl();
   const { data: currentEncounter } = useEncounter.document.byId(encounterId);
-
-  const { data: events } = useProgramEvents.document.list(
-    {
-      at: extractAt(currentEncounter, options),
-      patientId: currentEncounter?.patient?.id ?? '',
-      filter: {
-        type: {
-          equalTo: options?.eventType,
-        },
-        documentType: {
-          equalTo: currentEncounter?.type,
-        },
+  const patientId = config?.documentName?.split('/')[1];
+
+  const { data: events } = useProgramEvents.document.list({
+    at: currentEncounter ? extractAt(currentEncounter, options) : undefined,
+    patientId: currentEncounter
+      ? currentEncounter?.patient?.id
+      : patientId ?? '',
+    filter: {
+      type: {
+        equalTo: options?.eventType,
       },
-      page: {
-        first: 1,
+      documentType: {
+        equalTo: currentEncounter
+          ? currentEncounter?.type
+          : options?.documentType,
```

Comment:
Here we should also first try `documentType` and then the current document type. So the `documentType` from the config takes precedence over the the current document type. Maybe also good to add this in the config docs comment.

Reply:
I'm going to rename the control as well since it can be used generically now without programs

Reply:
Yes please rename. If you get the patient id from the document you can always get the documentType from this document.

The behaviour is:
If `documentType` is specified in the options only events for the specified `documentType` are displayed. If `documentType` is not specified the `documentType` and `documentName` of the current document are used.

Note if you want to display all events for the current `documentType` but not for the current `documentName` you would still have to specify the `documentType`...

For example:
- `documentType: CareProgram` can be used to display event types like `CareProgramStatus`
- `documentType: undefined` can be used to displays events types like `CurrentCareEncounterLabResult` (old use case)
- `documentType: CareEncounter` from within a CareEncounter document could be used to display event types like `OverallCareEncounterState`